### ISSUE TITLE
Javascript error occurred in the main process

### ISSUE THREAD

**Issue by 1fairyfox** [NebulousLabs/Sia-UI#207]
I came across this error on my Mac (OSX 64-bit). I started up Sia-UI and waited quite a bit for it to load, it indefinitely sat there at loading screen with swirling icon before, after 30 minutes, gave up and closed it after which this error popped up. Further attempts are consistent with indefinite loading and giving this error when closed.

I also began noticing this on my Windows PC (Windows 10 Pro 64-bit) and fixed it by entirely deleting the app and re-installing it and re-cyncing the blockchain but I only did that because I hadn't set it up yet so there were no wallets or files I needed to save. However my Mac is a different case and I'm afraid if its happened to 2 different computers it'll re-occur again.

Image of indefinite loading screen

![screen shot 2016-02-14 at 1 45 07 pm](https://cloud.githubusercontent.com/assets/1305564/13035992/54e3605e-d321-11e5-8332-1ced3b087513.png)

This is the error message my Mac gave out

```
Error: Object has been destroyed
    at Error (native)
    at signal (/Applications/Sia-UI.app/Contents/Resources/app/js/mainjs/initSiad.js:20:21)
    at null._repeat (/Applications/Sia-UI.app/Contents/Resources/app/js/mainjs/initSiad.js:34:3)
    at wrapper [as _onTimeout] (timers.js:275:19)
    at Timer.listOnTimeout (timers.js:92:15)
```

**Comment by 1fairyfox** [comment]
Well it just mysteriously started working on my Mac laptop, I gave the windows one 3 or 4 chances but I'll post here if it happens again and I get it fixed but what could cause this and what if I can't get it fixed next time.

**Comment by 1fairyfox** [comment]
Nevermind, I now have a new set of issues from this. After I got it working on my Mac by trying again, it loaded and I unlocked my wallet and noticed it said I had a zero balance with no transactions. I closed out of Sia-UI and re-opened it only to discover the loading indicator appear momentaritly before flashing to the number "1" and it just sits there. Screenshot attached.
![screen shot 2016-02-14 at 1 38 01 pm](https://cloud.githubusercontent.com/assets/1305564/13035936/4b804aa0-d320-11e5-9a48-1bab1080bf87.png)

And upon closing we're back to the main thread error noted above, screenshot attached as well

![screen shot 2016-02-14 at 1 41 19 pm](https://cloud.githubusercontent.com/assets/1305564/13035959/bc5f06e4-d320-11e5-9d89-ea54a0421669.png)

It now does this everytime, I guess the only other option is to delete the program entirely and re-install it but this can't become a habit, whats causing this?

**Comment by Mingling94** [comment]
Can you load zipped or tarred versions of your `Sia-UI` folder somewhere for us to inspect? Additionally, is your `Sia` folder outside of your `Sia-UI` folder? By default it is inside. You don't need to include that to send.

When your wallet seems to have reset itself, it's likely that the UI is pointed to a `Sia` folder that isn't what you were working with before. I suspect that the problem is in the UI mismanaging your Sia folder's location.

**Comment by TacoByte** [comment]
I've had this problem once, but I couldn't reproduce it so I let it go.

I managed to fix the issue by terminating siad which for some reason kept running after closing the UI. The "1" went away and the app started normally.

**Comment by DavidVorick** [comment]
Maybe something in the asynchronous handling is getting confused? Something's happening in the wrong order, causing the GUI to get confused?

I guess that doesn't match the behavior that restarting 'siad' works to fix the problem. Hmm. Siad is returning something (or failing to return something) consistently that's confusing the UI? Maybe it's the wrong version of siad?

**Comment by 1fairyfox** [comment]
@Mingling94 Attached is my Sia-UI app folder without the containing Sia folder. I uploaded it to the Sia network and attached is the zipped Sia file. The Sia folder wasn't outside the Sia-UI folder. I think the wallet reset itself as part of a bigger problem with communication to and from the siad daemon but I do think its from mismanagement.

@TacoByte I caught onto that shortly after posting the issue and inspected my running processes, sure enough the siad daemon was running in the background. I had the same thought and closed it but it just took me back to infinite loading screen. Upon closing there was no error but I never got it to work upon several re-opening tries however the "1" did go away.

@DavidVorick From speculation, it seems the siad included with Sia-UI is different from the siad included with the cli version in that its more buggy or different. I say that because, Sia-UI seems to try and manage siac and siad and upon closing is suppose to shutdown both. However siad keeps running in the background which appears to confuse Sia-UI.

Furthermore when I switched to the cli utilities with siad unclosed and still running in the background, siac was unable to connect to the daemon but upon closing siad leftover from Sia-UI and launching siad that was bundled with the cli utilities it connected just fine. Right now I'm successfully using the cli utilities (With its own blockchain and wallet seperate from the UI) but still unable to use the GUI.

Hope that helps

[Sia-UI.app.zip.sia.zip](https://github.com/NebulousLabs/Sia-UI/files/131059/Sia-UI.app.zip.sia.zip)

**Comment by TacoByte** [comment]
@DavidVorick I do remember playing around with different Sia versions/executables at the time. It's possible the UI got confused because it expected another version of siad running in the background.

**Comment by Mingling94** [comment]
A. @TacoByte Sia-UI actually doesn't check the version of siad at all (because it's packaged with a certain version of siad on release).

B. It also doesn't really use siac, so @wiseguy12851's comment about the cli working with its own siad but the leftover siad from Sia-UI not working spot on lays the fault on siad itself. Unless there's some way siad can remain running after Sia-UI closes and get funked by it enough to not work with siac. Sia-UI only closes siad with a `/daemon/stop` command for reference.

C. Lastly, the infinite loading screen tells us a lot. If you check `initSiad.js` line 33 and `loadingScreen.js` line 51, you'll see that the main process needs to constantly send a `loading` signal to the renderer process or else the screen would display `Siad has stopped responding :(` (Yes the smiley face too) **EDIT: the "1" or "0" is actually this message erroneously displayed, addressed in #209**. This could be one of two issues:
1. `SiadWrapper.start()` is not returning which brings us back to [this old issue about if setImmediate() is needed](https://github.com/NebulousLabs/Nodejs-Sia/commit/8fb20cab3781a04940e25bb9c164838d475ffb5e).
But because of point B, it's likely that:
2. Your siad's funky.
